src/mgr/DaemonServer.cc

```diff
@@ -428,6 +453,19 @@ bool DaemonServer::handle_report(MMgrReport *m)
     m->get_connection()->mark_down();
     session->put();
 
+    Mutex::Locker l(lock);
```

Comment:
The region where we want to hold the lock should be in a `{}` block

Reply:
thank you very much, i will fix it tomorrow